script.js

```diff
@@ -1,12 +1,284 @@
+
+  const allEpisodes =
+    getAllEpisodesFromAPI(initialURL).then(makePageForEpisodes);
+}
+
+async function getAllEpisodesFromAPI(url) {
+  let episodesStorage;
+
+  const response = await fetch(url);
+  episodesStorage = response.json();
+  return episodesStorage;
 }
 
 function makePageForEpisodes(episodeList) {
+  createHeader();
+  createShowsDropDownMenu();
+  loadNewShow();
+  createDropDownMenu(episodeList);
+  useDropdownToJumpToEpisode(episodeList);
+  createSearchbar();
+  createEpisodeCounter(episodeList);
+  createEpisodeCards(episodeList);
+  createFooter();
+}
+
+function createHeader() {
+  const rootElem = document.getElementById("root");
+  let pageHeaderElement = document.createElement("div");
+  pageHeaderElement.id = "page-header";
+  rootElem.append(pageHeaderElement);
+}
+
+function createShowsDropDownMenu() {
+  let dropDownMenuElement = document.createElement("select");
+  dropDownMenuElement.id = "show-select";
+  let defaultOption = document.createElement("option");
+  defaultOption.innerText = "Select Show";
+  dropDownMenuElement.append(defaultOption);
+
+  let allShows = getAllShows();
+
+  // good example of where to use a ternirary
+  allShows.sort(function (a, b) {
+    // if (a.name < b.name) {
+    //   return -1;
+    // }
+    // else {
+    //   return 1;
+    // }
+
+    return a.name < b.name ? 1 : -1;
+  });
+
+  for (const show of allShows) {
+    let dropDownOption = document.createElement("option");
+    //////////// Assign dropdown option values
+    dropDownOption.value = show.id;
+
+    dropDownOption.innerText = `${show.name}`;
+    dropDownMenuElement.append(dropDownOption);
+  }
+
+  let header = document.getElementById("page-header");
+  header.append(dropDownMenuElement);
+}
+
+function loadNewShow() {
+  let showDropDownMenu = document.getElementById("show-select");
+  let showDropDownChoice = showDropDownMenu.value;
+
+  showDropDownMenu.addEventListener("change", populateNewEpisodes);
+
+  function populateNewEpisodes() {
+    showDropDownChoice = showDropDownMenu.value;
+    let showURL = `https://api.tvmaze.com/shows/${showDropDownChoice}/episodes`;
+
+    // before making cards for new show, clear html
+    document.querySelector("#root").innerHTML = "";
+    getAllEpisodesFromAPI(showURL).then(makePageForEpisodes);
+  }
+}
+
+function createDropDownMenu(episodeList) {
+  // make a dropdown menu
+  // for loop on episode list to populate options in dropdown
+
+  let dropDownMenuElement = document.createElement("select");
+  dropDownMenuElement.id = "episode-select";
+  let defaultOption = document.createElement("option");
+  defaultOption.innerText = "Select Episode";
+  dropDownMenuElement.append(defaultOption);
+
+  for (const episode of episodeList) {
+    let dropDownOption = document.createElement("option");
+    //////////// Assign dropdown option values
+    dropDownOption.value = "EP" + episode.id;
+
+    // more padstart stuff -- let's put this in a function later
```

Comment:
Yes, making a function that creates an episode code given an episode object seems like a good idea! Especially given the code is repeated later. The principle DRY - Don't Repeat Yourself - is a good one to stick to. This is OK, but it's usually worth making a function if there's a meaningful operation in your code that you can give a good name to.